generators/import-jdl/index.js

```diff
@@ -201,63 +205,69 @@ module.exports = class extends BaseGenerator {
             this.debug('Building client');
             this.rebuildClient();
         }
+        if (this.applicationsLeftToGenerate.length !== 0) {
+            this.info(`Here are the application names to generate manually: ${this.applicationsLeftToGenerate.join(', ')}`);
+        }
+        if (this.entitiesLeftToGenerate.length !== 0) {
+            this.info(`Here are the entity names to generate manually: ${_.uniq(this.entitiesLeftToGenerate).join(', ')}`);
+        }
     }
 };
 
-function shouldGenerateApplications(generationConfiguration, jdlObject) {
-    return !generationConfiguration.options['ignore-application'] && Object.keys(jdlObject.applications).length !== 0;
-}
-
-function getApplicationPaths(jdlApplications) {
-    const paths = {};
-    Object.keys(jdlApplications).forEach((baseName) => {
-        paths[baseName] = jdlApplications[baseName].config.path;
-    });
-    return paths;
-}
-
-function setGeneratorVersionInJDLApplications(jdlApplications, version) {
-    Object.keys(jdlApplications).forEach((applicationName) => {
-        jdlApplications[applicationName].config.jhipsterVersion = version;
-    });
-    return jdlApplications;
+function shouldGenerateApplications(generator) {
+    return !generator.options['ignore-application'] && generator.importState.exportedApplications.length !== 0;
 }
 
 function generateApplicationFiles(generator, application) {
-    const initialWorkingDir = shelljs.pwd();
-    shelljs.cd(path.join(shelljs.pwd().stdout, application.path || ''));
-    callSubGenerator(generator, '..', 'app', {
-        force: generator.options.force,
-        debug: generator.options.debug,
-        'skip-install': generator.options['skip-install'],
-        'skip-user-management': application.config.skipUserManagement,
-        'jhi-prefix': application.config.jhiPrefix
+    const args = ['jhipster'];
+    if (generator.options.force) {
+        args.push('--force');
+    }
+    if (generator.options.debug) {
+        args.push('--debug');
+    }
+    if (generator.options['skip-install']) {
+        args.push('--skip-install');
+    }
+    if (application['generator-jhipster'].skipUserManagement) {
+        args.push('--skip-user-management');
+    }
+    if (application['generator-jhipster'].jhiPrefix) {
+        args.push('--jhi-prefix');
+        args.push(application['generator-jhipster'].jhiPrefix);
+    }
+    const done = generator.async();
+    generator.spawnCommand('yo', args).on('close', () => {
```

Comment:
I dont like this part TBH, but its fine for now, i'll rework this when I find time, as its quite tricky to get composing working correctly here

Reply:
I'm still trying to get it working when I have the time, but the main issue is passing the context from one generator to the other when composing. I thought it was automatic when composing generators, but I can't get it working.